### PR TITLE
chore: make generator version an optional param

### PR DESCRIPTION
**Pull request by JoeWang1127**
In this PR:
- Change `gapic_generator_version` as an optional parameter.
- Set `gapic_generator_version` through environment variable if it's not specified in the generation config.

### CODE REVIEW

**Review by blakeli0 — APPROVED** [review]
I was hoping we can skip calling the generator if the version is not provided, maybe reusing the `proto_only` flag, but it probably requires more effort to populate the value and pass the flag around. I think this approach if good for now.

File: .github/scripts/hermetic_library_generation.sh [hunk 1]
```diff
@@ -104,6 +107,7 @@ docker run \
   -u "$(id -u):$(id -g)" \
   -v "$(pwd):${workspace_name}" \
   -v "$HOME"/.m2:/home/.m2 \
+  -e generator_version="${generator_version}" \
```
Comment (diegomarquezp) [inline .github/scripts/hermetic_library_generation.sh:110]: From our [shell style guide](https://engdoc.corp.google.com/eng/doc/devguide/shell/styleguide.md?cl=head#s7.3-constants-and-environment-variable-names): _Constants and anything exported to the environment should be capitalized, separated with underscores, and declared at the top of the file._

The _top of the file_ part doesn't apply, but I think it's reasonable to make it uppercase for consistency with other env vars we set up in the Dockerfile

Comment (blakeli0) [reply .github/scripts/hermetic_library_generation.sh:110]: Makes sense, do you mind updating it in place of Joe?

Comment (diegomarquezp) [reply .github/scripts/hermetic_library_generation.sh]: Done

**Review by diegomarquezp — APPROVED** [review]

**Pull request merged**